### PR TITLE
Incorporate PiZeroAnalyzer code into ECAL DQM

### PR DESCRIPTION
**Pull request by alejands**
#### PR description:

ECAL DPG requested to have the offline DQM plots ([example offline GUI](https://tinyurl.com/2ykhlsat)) from `PiZeroAnalyzer` in `DQMOffline/EGamma` available in online DQM as well. 
- https://github.com/cms-sw/cmssw/blob/master/DQMOffline/EGamma/plugins/PiZeroAnalyzer.h
- https://github.com/cms-sw/cmssw/blob/master/DQMOffline/EGamma/plugins/PiZeroAnalyzer.cc
- https://github.com/cms-sw/cmssw/blob/master/DQMOffline/EGamma/python/piZeroAnalyzer_cfi.py

In order to do this, we have migrated the code above into a new ECAL DQM module `PiZeroTask` in `DQM/EcalMonitorTasks`, which can then create plots both online and offline.

This new code is functionally the same as the old analyzer but after discussing with DPG, we decided to leave the old analyzer code in to not break anything. Open to discussion.

Complimentary PR to dmwm/deployment:
- https://github.com/dmwm/deployment/pull/1155

#### PR validation:

Online DQM client run on test data. Offline DQM tested on relval WF 11634.511 (`runTheMatrix.py -w upgrade -l 11634.511`). In both cases, the new plots were created successfully and matched the features of the old plots.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Backports made to 12_4_X https://github.com/cms-sw/cmssw/pull/38570 and 12_3_X https://github.com/cms-sw/cmssw/pull/38571 to have this in time for the first 13.6 TeV stable collisions

### CODE REVIEW

**Comment by qliphy** [comment]
please test

**Comment by emanueleusai** [comment]
+1

**Comment by emanueleusai** [comment]
Can you please open the 12_4 and 12_3 backports, so we can deploy online?

File: DQM/EcalMonitorTasks/src/PiZeroTask.cc [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+        seleS4S9GammaTwo_(0.f),
+        selePtPi0_(0.f),
+        selePi0Iso_(0.f),
+        selePi0BeltDR_(0.f),
+        selePi0BeltDeta_(0.f),
+        seleMinvMaxPi0_(0.f),
+        seleMinvMinPi0_(0.f),
+        posCalcParameters_(edm::ParameterSet()) {}
+
+  void PiZeroTask::setParams(edm::ParameterSet const& params) {
```
Comment (perrotta) [inline DQM/EcalMonitorTasks/src/PiZeroTask.cc:22]: Were is this `setParams` method called? Should it be in the constructor? 

Can you also define a fillDescription method, to initialize and validated all those parameters?

Comment (alejands) [reply DQM/EcalMonitorTasks/src/PiZeroTask.cc:22]: > Were is this `setParams` method called? Should it be in the constructor?

The inheritance structure of ECAL DQM can be non-trivial at times. The ECAL DQM analyzer being run is `EcalDQMonitorTask`, which itself executes `DQWorker`s such as `PiZeroTask`. The analyzer parent class `EcalDQMonitor` constructor passes the parameters to the worker here:
https://github.com/cms-sw/cmssw/blob/c488deecab052ff8f5e4eeed6e122b55589de505/DQM/EcalCommon/src/EcalDQMonitor.cc#L35-L36
which are then set here:
https://github.com/cms-sw/cmssw/blob/c488deecab052ff8f5e4eeed6e122b55589de505/DQM/EcalCommon/src/DQWorker.cc#L194

Comment (alejands) [reply DQM/EcalMonitorTasks/src/PiZeroTask.cc:22]: > Can you also define a fillDescription method, to initialize and validate all those parameters?

`fillDescriptions()` is already called but is again non-trivial to see. The ECAL DQM analyzer has it's own `fillDescriptions()` which calls
https://github.com/cms-sw/cmssw/blob/c488deecab052ff8f5e4eeed6e122b55589de505/DQM/EcalMonitorTasks/plugins/EcalDQMonitorTask.cc#L85
which calls
https://github.com/cms-sw/cmssw/blob/c488deecab052ff8f5e4eeed6e122b55589de505/DQM/EcalMonitorTasks/src/DQWorkerTask.cc#L11
which finally calls
https://github.com/cms-sw/cmssw/blob/c488deecab052ff8f5e4eeed6e122b55589de505/DQM/EcalCommon/src/DQWorker.cc#L50

**Comment by perrotta** [comment]
+1
- Thank you @alejands for the explanations

**Pull request merged**